time/Cargo.toml

```diff
@@ -50,10 +50,20 @@ deranged = { workspace = true }
 itoa = { workspace = true, optional = true }
 quickcheck = { workspace = true, optional = true }
 rand = { workspace = true, optional = true }
-serde = { workspace = true, optional = true }
 time-core = { workspace = true }
 time-macros = { workspace = true, optional = true }
 
+# Serde:
+# <= 1.0.171 due to serde-rs/serde#2538
+# >= 1.0.175 due to serde-rs/serde#2590
+# No way to support multiple version ranges e.g. >= 1.0.126
+# Opt-in Allow precompiled
+[target.'cfg(precompiled = "allow")'.dependencies]
+serde = { version = ">= 1.0.126", default-features = false, optional = true }
```

Comment:
Alternatively, instead of `precompiled` there could be `older_serde` bound >= 1.0.126, <= 1.0.171`.

Reply:
Yeah but that restriction would be opt-in ? will have to check with the maintainer but I suspect it has to be allow-in

Reply:
By default, >= 185. By cfg, >= 126, < 172. Alternatively by cfg, >= 126.

One is for old serde, without precompiles. One is for old serde, including precompiles. One is for new serde.

That covers people who want to use old serde, yet don't want to allow precompiles.

Reply:
How about -
```
cfg(time_serde_versions = "all") -> ">=1.0.126"
cfg(time_serde_versions = "pre-171") -> ">1.0.126, <= 1.0.171"
default -> ">= 1.0.185"
```

Reply:
It's no longer explicit to being about precompiled, yet that would have the granularity I'm discussing. I'd let jhpratt decide what's optimal for time.